src/editor.cpp

```diff
@@ -2024,6 +2026,67 @@ void Editor::deleteEvent(Event *event) {
     //updateSelectedObjects();
 }
 
+void Editor::openMapScripts() const {
+    const QString scriptPath = project->getMapScriptsFilePath(map->name);
+    openInTextEditor(scriptPath);
+}
+
+void Editor::openScript(const QString &scriptLabel) const {
+    // Find the location of scriptLabel.
+    QStringList scriptPaths(project->getMapScriptsFilePath(map->name));
+    scriptPaths << project->getEventScriptsFilePaths();
+    int lineNum = 0;
+    QString scriptPath = scriptPaths.first();
+    for (const auto &path : scriptPaths) {
+        lineNum = ParseUtil::getScriptLineNumber(path, scriptLabel);
+        if (lineNum != 0) {
+            scriptPath = path;
+            break;
+        }
+    }
+
+    openInTextEditor(scriptPath, lineNum);
+}
+
+void Editor::openInTextEditor(const QString &path, int lineNum) const {
+    QString command = porymapConfig.getTextEditorGotoLine();
+    if (command.isEmpty()) {
+        // Open map scripts in the system's default editor.
+        QDesktopServices::openUrl(QUrl::fromLocalFile(path));
+    } else {
+        if (command.contains("%F")) {
+            if (command.contains("%L"))
+                command.replace("%L", QString::number(lineNum));
+            command.replace("%F", path);
+        } else {
+            command += ' ' + path;
+        }
+        startDetachedProcess(command);
+    }
+}
+
+void Editor::openProjectInTextEditor() const {
+    QString command = porymapConfig.getTextEditorOpenFolder();
+    if (command.contains("%D"))
+        command.replace("%D", project->root);
+    else
+        command += ' ' + project->root;
+    startDetachedProcess(command);
+}
+
+bool Editor::startDetachedProcess(const QString &command, const QString &workingDirectory, qint64 *pid) const {
+    logInfo("Executing command: " + command);
+#ifdef Q_OS_WIN
+    // On Windows, a QProcess command must be wrapped in a cmd.exe command.
+    const QString program = QProcessEnvironment::systemEnvironment().value("COMSPEC");
+    QStringList arguments({ QString("/c"), command });
+#else
+    QStringList arguments = QProcess::splitCommand(command);
```

Comment:
I'm having a lot of trouble getting this to work on Windows.  For example, this is failing:
```
"C:\Program Files\Sublime Text 3\sublime_text.exe" %F:%L
```
```
Executing command: "C:\Program Files\Sublime Text 3\sublime_text.exe" D:/cygwin64/home/huder/pokeemerald/data/maps/PetalburgCity/scripts.inc:103
'\"C:\Program Files\Sublime Text 3\sublime_text.exe\"' is not recognized as an internal or external command,
operable program or batch file.
```

Modifying the `startDetached()` call to simply `return QProcess::startDetached(command);` seems to make things simpler.  Is there any reason you opted for the 4-param version of `startDetached()`?

Also, `QProcess::splitCommand()` is a Qt 5.15 addition.

Reply:
`QProcess::startDetached(command)` has been deprecated which is why I avoided it and used the 4 param version, I could've left out the latter 2 params since they have default values, but I added them for the sake of code re-use. I can also remove `QProcess::splitCommand()`. I'll look into why the above example is failing.